### PR TITLE
Fixes #14333

### LINKED ISSUE #14333: Bun crashes when providing serverName on tls config
### How can we reproduce the crash?

```ts
Bun.serve({
	fetch: (request) => new Response('Welcome to Bun!'),
	port: 3000,
	tls: {
		cert: Bun.file('./certs/fullchain.pem'),
		key: Bun.file('./certs/privkey.pem'),
		serverName: 'somedomain.com'
	}
});

```

### Relevant log output

_No response_

### Stack Trace (bun.report)

Bun v1.1.29 ([`6d43b36`](<https://github.com/oven-sh/bun/tree/6d43b366220099042f28c5fa5c66185fe2580539>)) on macos aarch64 [AutoCommand]

Segmentation fault at address 0x00000178

- *1 unknown/js code*
- `us_bun_internal_ssl_socket_context_add_server_name`
- `us_bun_internal_ssl_socket_context_add_server_name`
- `us_bun_socket_context_add_server_name`
- `uws_add_server_name_with_options`
- `BunObject_callback_serve`
- *1 unknown/js code*
- `llint_entry`
- `vmEntryToJavaScript`
- `JSC::Interpreter::executeModuleProgram`

Features: jsc, Bun.stdin, dotenv, https\_server, tsconfig

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

Fixes #14333
Fixes https://github.com/oven-sh/bun/issues/9224

### How did you verify your code works?

There are tests

### CODE REVIEW

**Comment by robobun** [comment]
@Jarred-Sumner, your commit c84fa7594465496a6f4dca6f659fca9882e1a82f is building: [`#5057`](https://buildkite.com/bun/bun/builds/5057)

File: src/bun.js/api/server.zig [hunk 1]
```diff
@@ -1465,8 +1465,6 @@ pub const ServerConfig = struct {
                                 args.ssl_config = ssl_config;
                             } else {
                                 if (ssl_config.server_name == null or std.mem.span(ssl_config.server_name).len == 0) {
-                                    var config = ssl_config;
-                                    defer config.deinit();
```
Comment (dylan-conway) [inline src/bun.js/api/server.zig:1469]: This deinit looks safe to keep. It won't be assigned to `args.ssl_config` and it won't be appended to `args.sni`

File: src/bun.js/api/server.zig [hunk 2]
```diff
@@ -7082,81 +7079,138 @@ pub fn NewServer(comptime NamespaceType: type, comptime ssl_enabled_: bool, comp
...
+                    Bun__addInspector(ssl_enabled, app, this.globalThis);
                 }
 
-                this.app.get("/src:/*", *ThisServer, this, onSrcRequest);
+                app.get("/src:/*", *ThisServer, this, onSrcRequest);
             }
         }
 
         pub fn listen(this: *ThisServer) void {
             httplog("listen", .{});
```
Comment (dylan-conway) [inline src/bun.js/api/server.zig:7120]:
```suggestion
```

File: src/bun.js/api/server.zig [hunk 3]
```diff
@@ -7192,6 +7246,20 @@ pub fn NewServer(comptime NamespaceType: type, comptime ssl_enabled_: bool, comp
                     );
                 },
             }
+
+            if (globalThis.hasException()) {
+                this.deinit();
+                return;
+            }
+
+            this.ref();
```
Comment (dylan-conway) [inline src/bun.js/api/server.zig:7254]: what is the possible exception at this point?

**Review by dylan-conway — APPROVED** [review]

File: src/bun.js/api/server.zig (hunk 3 above)
Comment (Jarred-Sumner) [reply src/bun.js/api/server.zig:7254]: I"m not 100% sure ,but it felt worth checking

Comment (Jarred-Sumner) [reply src/bun.js/api/server.zig:7254]: onListenFailed

**Pull request merged**